scripts/solve_network.py

```diff
@@ -1093,6 +1093,10 @@ def solve_network(n, config, solving, **kwargs):
 
     # add to network for extra_functionality
     n.config = config
+    if "snakemake" in kwargs:
+        n.snakemake = kwargs.pop("snakemake")
+    else:
+        n.snakemake = snakemake
```

Comment:
I would suggest to follow the same structure as for `n.config`, i.e.:
```suggestion
    n.params = params
```